src/mito2/src/compaction/picker.rs

```diff
@@ -45,6 +47,82 @@ pub struct PickerOutput {
     pub time_window_size: i64,
 }
 
+/// SerializedPickerOutput is a serialized version of PickerOutput by replacing [CompactionOutput] and [FileHandle] with [SerializedCompactionOutput] and [FileMeta].
+#[derive(Default, Clone, Debug, Serialize, Deserialize)]
+pub struct SerializedPickerOutput {
+    pub outputs: Vec<SerializedCompactionOutput>,
+    pub expired_ssts: Vec<FileMeta>,
+    pub time_window_size: i64,
+}
+
+impl From<PickerOutput> for SerializedPickerOutput {
+    fn from(input: PickerOutput) -> Self {
+        let outputs = input
+            .outputs
+            .into_iter()
+            .map(|output| SerializedCompactionOutput {
+                output_file_id: output.output_file_id,
+                output_level: output.output_level,
+                inputs: output
+                    .inputs
+                    .into_iter()
+                    .map(|s| s.meta_ref().clone())
+                    .collect(),
+                filter_deleted: output.filter_deleted,
+                output_time_range: output.output_time_range,
+            })
+            .collect();
+
+        let expired_ssts = input
+            .expired_ssts
+            .into_iter()
+            .map(|s| s.meta_ref().clone())
+            .collect();
+
+        Self {
+            outputs,
+            expired_ssts,
+            time_window_size: input.time_window_size,
+        }
+    }
+}
```

Comment:
nit: Taking the reference of `PickerOutput` is enough so that you don't have to consume it:

```suggestion
impl From<&PickerOutput> for SerializedPickerOutput {
    fn from(input: &PickerOutput) -> Self {
        let outputs = input
            .outputs
            .iter()
            .map(|output| SerializedCompactionOutput {
                output_file_id: output.output_file_id,
                output_level: output.output_level,
                inputs: output.inputs.iter().map(|s| s.meta_ref().clone()).collect(),
                filter_deleted: output.filter_deleted,
                output_time_range: output.output_time_range,
            })
            .collect();

        let expired_ssts = input
            .expired_ssts
            .iter()
            .map(|s| s.meta_ref().clone())
            .collect();

        Self {
            outputs,
            expired_ssts,
            time_window_size: input.time_window_size,
        }
    }
}

```